tools/RAiDER/aria/prepFromGUNW.py

```diff
@@ -5,7 +5,268 @@
 # RESERVED. United States Government Sponsorship acknowledged.
 #
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+import os
+from datetime import datetime
+import numpy as np
+import xarray as xr
+import rasterio
+import yaml
+import shapely.wkt
+import RAiDER
+from RAiDER.utilFcns import rio_open, writeArrayToRaster
+from RAiDER.logger import logger
+from eof.download import download_eofs
+
+## ToDo:
+    # Heights
+    # Capture metadata
+
+## cube spacing in degrees for each model
+DCT_POSTING = {'HRRR': 0.01, 'HRES': 0.05, 'GMAO': 0.1, 'ERA5': 0.1}
+
+## these 2 are not used ---------
+def my_ceil(a, precision=0):
+    ## round down to nearest 'precision'
+    return np.true_divide(np.ceil(a * 10**precision), 10**precision)
+
+
+def my_floor(a, precision=0):
+    ## round down to nearest 'precision'
+    return np.true_divide(np.floor(a * 10**precision), 10**precision)
+## ------------------------------
+
+
+class GUNW(object):
+    def __init__(self, f:str, wm:str, out_dir:str):
+        self.path_gunw = f
+        self.wmodel    = wm
+        self.out_dir   = out_dir
+        self.spacing_m = int(DCT_POSTING[self.wmodel] * 1e5)
+
+
+    def __call__(self):
+        self.SNWE      = self.get_bbox()
+        self.heights   = np.arange(-500, 9500, 500).tolist()
+        self.dates     = self.get_dates()
+        self.ref_time  = self.get_time()
+        self.look_dir  = self.get_look_dir()
+        self.wavelength = self.get_wavelength()
+        self.name       = f'{self.dates[0]}-{self.dates[1]}'
+        self.OrbitFiles = self.get_orbit_files()
+
```

Comment:
It looks like you should be using the `@property` might be helpful here that way properties are computed on the fly.